aries_cloudagent/wallet/sd_jwt.py

```diff
@@ -0,0 +1,306 @@
+"""Operations supporting SD-JWT creation and verification."""
+
+import json
+import re
+from typing import Any, List, Mapping, Optional
+from marshmallow import fields
+from jsonpath_ng.ext import parse
+from sd_jwt.common import SDObj
+from sd_jwt.issuer import SDJWTIssuer
+from sd_jwt.verifier import SDJWTVerifier
+
+from ..core.profile import Profile
+from ..wallet.jwt import JWTVerifyResult, JWTVerifyResultSchema, jwt_sign, jwt_verify
+from ..core.error import BaseError
+from ..messaging.valid import StrOrDictField
+
+
+CLAIMS_NEVER_SD = ["iss", "iat", "exp", "cnf"]
+
+
+class SDJWTError(BaseError):
+    """SD-JWT Error."""
+
+
+class SDJWTIssuerACAPy(SDJWTIssuer):
+    """SDJWTIssuer class for ACA-Py implementation."""
+
+    def __init__(
+        self,
+        user_claims: dict,
+        issuer_key,
+        holder_key,
+        profile: Profile,
+        headers: dict,
+        did: Optional[str] = None,
+        verification_method: Optional[str] = None,
+        add_decoy_claims: bool = False,
+        serialization_format: str = "compact",
+    ):
+        """Initialize an SDJWTIssuerACAPy instance."""
+        self._user_claims = user_claims
+        self._issuer_key = issuer_key
+        self._holder_key = holder_key
+
+        self.profile = profile
+        self.headers = headers
+        self.did = did
+        self.verification_method = verification_method
+
+        self._add_decoy_claims = add_decoy_claims
+        self._serialization_format = serialization_format
+        self.ii_disclosures = []
+
+    async def _create_signed_jws(self):
+        self.serialized_sd_jwt = await jwt_sign(
+            self.profile,
+            self.headers,
+            self.sd_jwt_payload,
+            self.did,
+            self.verification_method,
+        )
+
+    async def issue(self):
+        """Issue an sd-jwt."""
+        self._check_for_sd_claim(self._user_claims)
+        self._assemble_sd_jwt_payload()
+        await self._create_signed_jws()
+        self._create_combined()
+
+
+def create_json_paths(it, current_path="", path_list=None):
+    """Create a json path for each element of the payload."""
+    if path_list is None:
+        path_list = []
+
+    if type(it) is dict:
```

Comment:
Type checking should be done using `isinstance()`.
```suggestion
    if isinstance(it, dict):
```